shotgun_api3/shotgun.py

```diff
@@ -1132,6 +1132,31 @@ def _add_project_param(self, params, project_entity):
             params["project"] = project_entity
 
         return params
+    
+    def _translate_update_params(
+        self, entity_type, entity_id, data, multi_entity_update_modes
+    ):
+        global SHOTGUN_API_ENABLE_ENTITY_OPTIMIZATION
+
+        def optimize_value(field_value):
+            if isinstance(field_value, dict) and SHOTGUN_API_ENABLE_ENTITY_OPTIMIZATION:
```

Comment:
Isn't it better?
```
if SHOTGUN_API_ENABLE_ENTITY_OPTIMIZATION and isinstance(field_value, dict): 
   ...
```
cause if SHOTGUN_API_ENABLE_ENTITY_OPTIMIZATION is normally False, this form might perform an unnecessary evaluation of isinstance(field_value, dict).

Reply:
Good call. I'll do that suggestion